packages/app/src/systems/Send/machines/sendMachine.ts

```diff
@@ -0,0 +1,298 @@
+/* eslint-disable @typescript-eslint/consistent-type-imports */
+import { toast } from '@fuel-ui/react';
+import type { Asset } from '@fuel-wallet/types';
+import {
+  BN,
+  ScriptTransactionRequest,
+  TransactionResponse,
+  WalletUnlocked,
+} from 'fuels';
+import { createMachine, InterpreterFrom, send, StateFrom } from 'xstate';
+
+import {
+  AccountInputs,
+  UnlockMachine,
+  unlockMachine,
+  unlockMachineErrorAction,
+  UnlockMachineEvents,
+} from '~/systems/Account';
+import {
+  assignWithData,
+  assignWithInput,
+  assignErrors,
+  ChildrenMachine,
+  FetchMachine,
+  resetContext,
+} from '~/systems/Core';
+import { GroupedErrors } from '~/systems/Transaction';
+import { TxService } from '~/systems/Transaction/services';
+
+export enum SendScreens {
+  unlocking,
+  select,
+  confirm,
+}
+
+type MachineContext = {
+  inputs?: {
+    asset?: Asset;
+    address?: string;
+    amount?: BN;
+    wallet?: WalletUnlocked;
+  };
+  response?: {
+    fee?: BN;
+    txRequest?: ScriptTransactionRequest;
+    txApprove?: TransactionResponse;
+  };
+  errors?: {
+    unlockError?: string;
+    txRequestError?: GroupedErrors;
+    txApproveError?: GroupedErrors;
+  };
+};
+
+type Inputs = MachineContext['inputs'];
+
+type TxRequestReturn = {
+  fee: BN;
+  txRequest: ScriptTransactionRequest;
+};
+
+type TxApproveReturn = {
+  txApprove: TransactionResponse;
+};
+
+type MachineServices = {
+  unlock: {
+    data: WalletUnlocked;
+  };
+  createTx: {
+    data: TxRequestReturn;
+  };
+  sendTx: {
+    data: TxApproveReturn;
+  };
+};
+
+type MachineEvents =
+  | { type: 'SET_ASSET'; input: Asset }
+  | { type: 'SET_ADDRESS'; input: string }
+  | { type: 'SET_AMOUNT'; input: BN }
+  | { type: 'UNLOCK_WALLET'; input: AccountInputs['unlock'] }
+  | { type: 'RESET'; input: null }
+  | { type: 'BACK'; input: null }
+  | { type: 'CONFIRM'; input: null };
+
+export const sendMachine = createMachine(
+  {
+    predictableActionArguments: true,
+    tsTypes: {} as import('./sendMachine.typegen').Typegen0,
+    schema: {
+      context: {} as MachineContext,
+      events: {} as MachineEvents,
+      services: {} as MachineServices,
+    },
+    id: '(machine)',
+    initial: 'unlocking',
+    states: {
+      unlocking: {
+        invoke: {
+          id: 'unlock',
+          src: 'unlock',
+          data: (_: MachineContext, ev: MachineEvents) => ev.input,
+          onDone: [
+            unlockMachineErrorAction('unlocking', 'unlockError'),
+            { target: 'idle', actions: 'assignWallet' },
+          ],
+        },
+        on: {
+          UNLOCK_WALLET: {
+            actions: [
+              send<MachineContext, UnlockMachineEvents>(
+                (_, ev) => ({ type: 'UNLOCK_WALLET', input: ev.input }),
+                { to: 'unlock' }
+              ),
+            ],
+          },
+        },
+      },
+      idle: {
```

Comment:
this `idle` is not actually `idle` as it does a lot of things in substates...

should be same style of `confirming` state, that has a first `idle` substate

Reply:
✅